controllers/storagecluster/provider_server.go

```diff
@@ -46,21 +45,19 @@ func (o *ocsProviderServer) ensureCreated(r *StorageClusterReconciler, instance
 
 	r.Log.Info("Spec.AllowRemoteStorageConsumers is enabled. Creating Provider API resources")
 
-	var finalErr error
-
-	for _, f := range []func(*StorageClusterReconciler, *ocsv1.StorageCluster) error{
+	for _, f := range []func(*StorageClusterReconciler, *ocsv1.StorageCluster) (reconcile.Result, error){
 		o.createSecret,
 		o.createService,
 		o.createDeployment,
 	} {
-		err := f(r, instance)
-		if err != nil {
-			multierr.AppendInto(&finalErr, err)
-			continue
+		res, err := f(r, instance)
+
+		if !res.IsZero() || err != nil {
+			return res, err
 		}
 	}
 
-	return reconcile.Result{}, finalErr
```

Comment:
Why complicate things with a loop? 
You force every create func to have the exact same sig! no args and 2 return values. Why put these restrictions on the developer?

```suggestion
	if err := o.createSecret; err != nil {
		return reconcile.Result{}, err
	}
	
	if result, err: = o.createService; err != nil {
		return err
	} else if !res.IsZero() || {
		return result, nil
	} 
	
	if err := o.createDeployment; err != nil {
		return reconcile.Result{}, err
	}
	
	return reconcile.Result{}, nil
 ```

Reply:
done